### ISSUE TITLE
Support for systemd timers

### ISSUE THREAD

**Issue by krieghof** [teejee2008/timeshift#201]
Some distributions don't use cron they use systemd timers. 
For example Solus, on Solus we have to run `timeshift --check` every now and then to ensure that the snapshots are taken.

**Comment by teejee2008** [comment]
It's not installed by default, but you can install cron on Solus. There are no plans to add support for systemd timers.

**Comment by krieghof** [comment]
On Solus I can install dcron. But somehow it does not work.

In settings I activated the snapshot on boot and the hourly snapshots, then i restarted my PC and waited 2 hours but no snapshots.
Is this a problem with dcron on Solus?

I want to package timeshift for Solus, because I think timeshift is a must have on every distro.
Thanks for your help so far.

**Comment by teejee2008** [comment]
I'll have to install Solus to test it out. This may take some time.

**Comment by Tids** [comment]
even if cron is in distro XY, it would be good to have support for systemd timers. At work we have replaced almost 3000 cron entries (over the last 2 years) with systemd timers. On my personal server/desktop I also don't use cron anymore.

**Comment by Astilex5** [comment]
I just wanted to mention that Solus has fully forgone any cron based package. dcron, cronie, and cron are all refused and any mention of it is met with silence or lashback.

The official solus stance on cron: "Use systemd timers".

It would be nice if there was an option to at least build this without cron support. As timeshift is really the only backup tool that makes solid sense.

**Comment by Skylead** [comment]
dcron was purged from Solus last year https://dev.getsol.us/T6338, any update on looking into refactoring with systemd timers? Or maybe make it an optional dependency and just disable scheduling if it's not in the system so people can use it for manual snapshots

**Comment by Astilex5** [comment]
I agree with @Skylead. Not so much with the Use of Systemd timers as Any non systemd distro (Such as Void, Devuan, and Gentoo) would lose out. I'm more on the side of making cron an optional Dep so it allows the use on Distros that do not ship cron. This way there will be minimal breakage.

**Comment by teejee2008** [comment]
Please submit a PR

**Comment by koocotte** [comment]
See #787
